src/victory-primitives/clip-path.js

```diff
@@ -58,16 +56,16 @@ export default class ClipPath extends React.Component {
   calculateAttributes(props) {
     const { clipWidth, clipHeight, translateX, translateY } = props;
     const padding = Helpers.getPadding(props);
-    const clipPadding = props.clipPadding || {};
+    const clipPadding = Helpers.getPadding({ padding: props.clipPadding });
+
     const totalPadding = (side) => {
-      const total = +padding[side] - (clipPadding[side] || 0);
-      return typeof total === "number" ? total : 0;
+      return padding[side] - clipPadding[side];
     };
     return {
-      x: totalPadding("left") + (translateX || 0),
-      y: totalPadding("top") + (translateY || 0),
-      width: Math.max(+clipWidth - totalPadding("left") - totalPadding("right"), 0),
-      height: Math.max(+clipHeight - totalPadding("top") - totalPadding("bottom"), 0)
+      x: totalPadding("left") + translateX,
+      y: totalPadding("top") + translateY,
+      width: Math.max(clipWidth - totalPadding("left") - totalPadding("right"), 0),
```

Comment:
clipWidth is already being validated as numerical though. Is there a reason we'd be unsure of type here and require a cast? 

I'd like to hear your thoughts here. Usually I prefer to enforce types once and then have full confidence afterwards. Multiple layers of type checks tend to make me worry that there's more complication and edge cases than may actually exist.